### DISCUSSION TITLE
[Edge runtime] Unclear how long waitUntil can last

### DISCUSSION THREAD

**Discussion by louis030195** [vercel/community#2513] [Help] [answered]
### Summary

According to

https://vercel.com/docs/concepts/functions/edge-middleware/middleware-api#waituntil

>Prolongs the execution of the function until the promise passed to `waitUntil` is resolved

It does not say if my promise can last 2 years or 3 minutes?

**Comment by amyegan** [comment, marked as answer]
Hi @louis030195! You can find more information on [Maximum Execution Duration](https://vercel.com/docs/concepts/functions/edge-middleware/limitations#maximum-execution-duration), [Invocation limits](https://vercel.com/docs/concepts/functions/edge-middleware/usage-and-pricing#invocation-limits), and [CPU time limits](https://vercel.com/docs/concepts/functions/edge-middleware/usage-and-pricing#cpu-time-limits) in the [Edge Middleware Limitations](https://vercel.com/docs/concepts/functions/edge-middleware/limitations#) and [Usage & Pricing for Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware/usage-and-pricing#) documentation

**Reply by ShantanuNair** [reply]
@amyegan That's not very helpful unfortunately, at least in my case of a similar query, and I've gone through those documentation pages. My query is regarding a waitUntil that runs say 30 minutes of post response processing (yes, it will be under the CPU Time limit, just taking long wall clock time). I can't figure out a yes/no to this from the documents you give. This is assuming edge functions, with streaming, and the client has disonnected. an you please help clear this up?

**Reply by smaeda-ks** [reply]
@ShantanuNair 30 minutes background task using `waitUntil` is definitely not going to work.
While there's no set limit, after ~30 seconds there is a slightly higher chance of eviction and your background task completion is not guaranteed.

**Reply by vincentdnl** [reply]
I feel like it's not really clear how long `waitUntil` is supposed to last and what these "chances of eviction" are. I have longer processes and I have no way to know if they are going to work or not.

**Reply by ShantanuNair** [reply]
@smaeda-ks Well well, https://blog.cloudflare.com/workers-pricing-scale-to-zero/ Will this change things? Would be a pleasant surprise :)
